### PR TITLE
🔧 MAINTAIN: Updated parser docstrings

### PR DESCRIPTION
**Pull request by ZviBaratz**
I was trying to get a little familiar with the code and thought I would use the opportunity to propose completed docstrings for the `parser` module. If this sort of contribution is welcome, I could try and do the same for the other modules.

### CODE REVIEW

**Comment by chrisjsewell** [comment]
always welcome 😄

File: sphinx_external_toc/parsing.py [hunk 1]
```diff
@@ -29,26 +29,32 @@
...
+    items_keys: Sequence[str] = attr.ib(default=())
+    default_subtrees_key: str = attr.ib(default=DEFAULT_SUBTREES_KEY)
+    default_items_key: str = attr.ib(default=DEFAULT_ITEMS_KEY)
 
     def get_subtrees_key(self, depth: int) -> str:
-        """Get the subtrees key name for this depth in the ToC
+        """Get the subtrees key name for this depth in the ToC.
 
         :param depth: recursive depth (starts at 0)
+        :type depth: str
```
Comment (chrisjsewell) [inline sphinx_external_toc/parsing.py:42]: `type`/`rtype` are not necessary in the docstring, since they are provided by the type annotations

**Review by chrisjsewell — CHANGES_REQUESTED** [review]
yeh just I think the types should not be duplicated in the docstring (plus they can be properly tested by mypy with the type annotations)

File: sphinx_external_toc/parsing.py (hunk 1 above)
Comment (ZviBaratz) [reply sphinx_external_toc/parsing.py:42]: Gotcha

**Review by chrisjsewell — APPROVED** [review]
Nice one!

**Pull request merged**